v2/datastream-to-spanner/terraform/samples/postgresql-end-to-end/README.md

```diff
@@ -0,0 +1,603 @@
+## Sample Scenario: PostgreSQL to Spanner production live migration using PostgreSQL source configuration
+
+> **_SCENARIO:_** This Terraform example illustrates launching a live migration
+> job for a PostgreSQL source, setting up all the required cloud infrastructure.
+> **Only the source details are needed as input.**
+
+## Terraform permissions
+
+In order to create the resources in this sample,
+the`Service account`/`User account` being used to run Terraform
+should have the
+required [permissions](https://cloud.google.com/iam/docs/manage-access-service-accounts#multiple-roles-console).
+There are two ways to add permissions -
+
+1. Adding pre-defined roles to the service account running Terraform.
+2. Creating a custom role with the granular permissions and attaching it to the
+   service account running Terraform.
+
+### Using custom role and granular permissions (recommended)
+
+Following permissions are required - 
+
+```shell
+- compute.globalAddresses.create
+- compute.globalAddresses.createInternal
+- compute.globalAddresses.delete
+- compute.globalAddresses.deleteInternal
+- compute.globalAddresses.get
+- compute.globalOperations.get
+- compute.networks.addPeering
+- compute.networks.get
+- compute.networks.listPeeringRoutes
+- compute.networks.removePeering
+- compute.networks.use
+- compute.routes.get
+- compute.routes.list
+- compute.subnetworks.get
+- compute.subnetworks.list
+- dataflow.jobs.cancel
+- dataflow.jobs.create
+- dataflow.jobs.updateContents
+- datastream.connectionProfiles.create
+- datastream.connectionProfiles.delete
+- datastream.privateConnections.create
+- datastream.privateConnections.delete
+- datastream.streams.create
+- datastream.streams.delete
+- datastream.streams.update
+- iam.roles.get
+- iam.serviceAccounts.actAs
+- pubsub.subscriptions.create
+- pubsub.subscriptions.delete
+- pubsub.topics.attachSubscription
+- pubsub.topics.create
+- pubsub.topics.delete
+- pubsub.topics.getIamPolicy
+- pubsub.topics.setIamPolicy
+- resourcemanager.projects.setIamPolicy
+- storage.buckets.create
+- storage.buckets.delete
+- storage.buckets.update
+- storage.objects.delete
+```
```

Comment:
A couple of permissions were added here, could you refer to this PR and include the changes to this template as well - 

https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1802

Reply:
Done